mypy.ini

```diff
@@ -515,6 +515,9 @@ ignore_errors = True
 [mypy-scipy.optimize.minpack2]
 ignore_errors = True
 
+[mypy-scipy.optimize.tests.test_milp]
+ignore_errors = True
```

Comment:
@Smit-create 

I was getting this error on the Meson build:
```
~/work/scipy/scipy/tools ~/work/scipy/scipy
Traceback (most recent call last):
  File "/home/runner/work/scipy/scipy/tools/check_installation.py", line 74, in <module>
    main(install_dir)
  File "/home/runner/work/scipy/scipy/tools/check_installation.py", line 49, in main
    raise Exception("%s is not installed" % scipy_test_files[test_file])
Exception: /home/runner/work/scipy/scipy/scipy/optimize/tests/test_milp.py is not installed
Error: Process completed with exit code 1.
```

I saw you make a similar change in gh-15120. What is this for? Should my change here fix the error above?

In `setup.py`, we already have
```
    config.add_data_dir('tests')
```
in `setup.py`.
Of the first two changes in gh-15120, that's what I might imagine referring to as "installing". Is the change to `mypy.ini` also considered part of "installing"?

Reply:
The failure is because meson is unable to find `test_milp.py`. You need to add the file name in the list here https://github.com/scipy/scipy/blob/main/scipy/optimize/tests/meson.build

Reply:
Thanks! Should I get rid of the change to mypy.ini?

Reply:
And is there somewhere else I need to add `_milp.py`? Now I'm getting e.g.
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/runner/work/scipy/scipy/installdir/lib/python3.9/site-packages/scipy/integrate/__init__.py", line 96, in <module>
    from ._bvp import solve_bvp
  File "/home/runner/work/scipy/scipy/installdir/lib/python3.9/site-packages/scipy/integrate/_bvp.py", line 9, in <module>
    from scipy.optimize import OptimizeResult
  File "/home/runner/work/scipy/scipy/installdir/lib/python3.9/site-packages/scipy/optimize/__init__.py", line 428, in <module>
    from ._milp import milp
ModuleNotFoundError: No module named 'scipy.optimize._milp'
```

Reply:
Yeah, that's true. It works.

Reply:
> Should I get rid of the change to mypy.ini?

Yeah, I think if there are no mypy failures, you can. It wasn't actually the reason for meson failure.

Reply:
Thanks, I'll do that. I think I had individually silenced the mypy issues already.